security/pkg/server/ca/server.go

```diff
@@ -117,17 +117,42 @@ func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertifi
 	return response, nil
 }
 
+func recordCertsExpiry(ca CertificateAuthority) {
+	caKeyCertBundler := ca.GetCAKeyCertBundle()
+	rootCertExpiryTimestamp.Record(extractRootCertExpiryTimestamp(caKeyCertBundler))
+
+	if len(caKeyCertBundler.GetCertChainPem()) > 0 {
+		certChainExpiryTimestamp.Record(extractCertChainExpiryTimestamp(caKeyCertBundler))
+	}
+}
+
 // extractRootCertExpiryTimestamp returns the unix timestamp when the root becomes expires.
-func extractRootCertExpiryTimestamp(ca CertificateAuthority) float64 {
-	rb := ca.GetCAKeyCertBundle().GetRootCertPem()
+func extractRootCertExpiryTimestamp(keyCertBundle util.KeyCertBundle) float64 {
+	rb := keyCertBundle.GetRootCertPem()
 	cert, err := util.ParsePemEncodedCertificate(rb)
 	if err != nil {
 		serverCaLog.Errorf("Failed to parse the root cert: %v", err)
 		return -1
 	}
 	end := cert.NotAfter
 	if end.Before(time.Now()) {
-		serverCaLog.Errorf("Expired Citadel Root found, x509.NotAfter %v, please transit your root", end)
+		serverCaLog.Errorf("Expired root cert found, x509.NotAfter %v, please transit your root cert", end)
+	}
+	return float64(end.Unix())
+}
+
+// extractCertChainExpiryTimestamp returns the unix timestamp when the cert chain becomes expires.
+func extractCertChainExpiryTimestamp(keyCertBundle util.KeyCertBundle) float64 {
+	cc := keyCertBundle.GetCertChainPem()
+	cert, err := util.ParsePemEncodedCertificate(cc)
```

Comment:
I think you are only parsing the the first cert in the cert chain if it contains multiple certs. That's fine but we should make it clear in the comment.